src/natcap/invest/sdr/sdr.py

```diff
@@ -1073,16 +1054,24 @@ def ls_factor_function(
         # avg aspect intermediate output should always have a defined
         # nodata value from pygeoprocessing
         valid_mask = (
-            (~utils.array_equals_nodata(avg_aspect, avg_aspect_nodata)) &
             ~utils.array_equals_nodata(percent_slope, slope_nodata) &
             ~utils.array_equals_nodata(
                 flow_accumulation, flow_accumulation_nodata))
         result = numpy.empty(valid_mask.shape, dtype=numpy.float32)
         result[:] = _TARGET_NODATA
 
-        contributing_area = (flow_accumulation[valid_mask]-1) * cell_area
+        # Although Desmet & Govers (1996) discusses "upstream contributing
+        # area", this is not strictly defined. We decided to use the square
+        # root of the upstream contributing area here as an estimate, which
+        # matches the SAGA LS Factor option "square root of catchment area".
+        # See the InVEST ADR-0001 for more information.
+        contributing_area = numpy.sqrt(
+            (flow_accumulation[valid_mask]-1) * cell_area)
```

Comment:
I'm not sure I understand the purpose of the `-1`. Is it that `flow_accumulation` includes each pixel itself as well as its upstream pixels, and we only want the upstream pixels?